components/notebook-controller/go.mod

```diff
@@ -1,78 +1,87 @@
 module github.com/kubeflow/kubeflow/components/notebook-controller
 
-go 1.23.0
+go 1.24.0
 
-toolchain go1.23.6
+toolchain go1.24.6
 
 require (
-	github.com/go-logr/logr v1.4.1
-	github.com/kubeflow/kubeflow/components/common v0.0.0-20220218084159-4ad0158e955e
+	github.com/go-logr/logr v1.4.3
+	github.com/kubeflow/kubeflow/components/common v0.0.0-20250812202812-6dbe51613019
 	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.30.0
-	github.com/prometheus/client_golang v1.18.0
-	k8s.io/api v0.29.0
-	k8s.io/apimachinery v0.29.0
-	k8s.io/client-go v0.29.0
-	sigs.k8s.io/controller-runtime v0.17.0
+	github.com/onsi/gomega v1.36.1
+	github.com/prometheus/client_golang v1.23.2
+	k8s.io/api v0.34.2
+	k8s.io/apimachinery v0.34.2
+	k8s.io/client-go v0.34.2
+	sigs.k8s.io/controller-runtime v0.22.4
```

Comment:
I'm slightly concerned about these k8s updates. We should be sure we're not breaking compatibility with OCP 4.16 as RHOAI 2.22 is still supporting it officially: https://access.redhat.com/support/policy/updates/rhoai-sm/lifecycle.